<H1>ACTIVTIY1: PALMER PENGUINS DATA ANALYTICS</H1>
<HR>
<H3>ANALYST: MARIA ASHLEY JOY ENALLO</H3>

In [1]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

In [2]:
penguin = pd.read_csv('dataset\\penguins.csv')

In [3]:
penguin

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen            39.1           18.7   
1        2     Adelie  Torgersen            39.5           17.4   
2        3     Adelie  Torgersen            40.3           18.0   
3        4     Adelie  Torgersen             NaN            NaN   
4        5     Adelie  Torgersen            36.7           19.3   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream            55.8           19.8   
340    341  Chinstrap      Dream            43.5           18.1   
341    342  Chinstrap      Dream            49.6           18.2   
342    343  Chinstrap      Dream            50.8           19.0   
343    344  Chinstrap      Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2                195.0       3250.0  female  2007  
3                  NaN          NaN     NaN  2007  
4                193.0       3450.0  female  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[344 rows x 9 columns]

<h2>QUESTION#1: HOW MANY FEMALE BORN IN 2007?</h2>
<HR>

In [7]:
year_female = penguin.groupby(['sex','year']).size()

In [8]:
year_female

sex     year
female  2007    51
        2008    56
        2009    58
male    2007    52
        2008    57
        2009    59
dtype: int64

<H2>INSIGHT: BASED ON THE DATA ABOVE THERE ARE 51 FEMALE PENGUIN BORN IN 2007</h2>
<HR>

<h2>QUESTION#2: HOW MANY MALE BORN IN DREAM ISLAND?</h2>
<HR>

In [16]:
male_dream = penguin.groupby(['island', 'sex']).size()

In [17]:
male_dream

island     sex   
Biscoe     female    80
           male      83
Dream      female    61
           male      62
Torgersen  female    24
           male      23
dtype: int64

<h2>INSIGHT: THERE ARE 62 MALE BORN/LIVE IN DREAM ISLAND</h2>
<HR>

<h2>QUESTION#3: IS THERE A CORRELATION IN PENGUINS FLIPPER AND BODY MASS?</h2>
<HR>

In [18]:
flipper_mass = penguin['flipper_length_mm'].corr(penguin['body_mass_g'])

In [19]:
flipper_mass

np.float64(0.8712017673060116)

<h2>INSIGHT: THEREFORE, THERE IS A HIGH CORRELATION BETWEEN PENGUINS FLIPPER AND BODY MASS</h2>
<HR>

<h2>QUESTION#4: AVERAGE BILL_LENGTH_MM MALES</h2>
<HR>

In [22]:
penguin[penguin['sex'] == 'male']['bill_length_mm'].mean()

np.float64(45.85476190476191)

<h2>INSIGHT: THERE IS AN AVERAGE OF 45.85476190476191 IN MALES BILL LENGTH</h2>
<HR>

<h2>QUESTION#5: WHAT SPECIES COMMONLY FOUND IN TORGERSEN?</h2>
<HR>

In [25]:
penguin[penguin['island'] == 'Torgersen']['species'].value_counts()

species
Adelie    52
Name: count, dtype: int64

<h2>INSIGHT: THE SPECIES COMMONLY FOUND IN TORGERSEN IS ADELIE</h2>
<HR>

<h2>QUESTION#6: HOW MANY PENGUINS BORN/ALIVE IN YEAR 2009?</h2>
<HR>

In [26]:
year = penguin.groupby('year').size()

In [27]:
year

year
2007    110
2008    114
2009    120
dtype: int64

<h2>INSIGHT: 120 PENGUINS BORN OR ALIVE IN YEAR 2009</h2>
<HR>

<h2>QUESTION#7: AVERAGE BODY MASS IN BISCOE</h2>
<HR>

In [28]:
penguin[penguin['island'] == 'Biscoe']['body_mass_g'].mean()

np.float64(4716.017964071856)

<h2>INSIGHT: THE AVERAGE MASS OF PENGUINS IN BISCOE IS 4716.017964071856</h2>
<HR>

<h2>QUESTION#8: HOW MANY FEMALE PENGUINS LIVE IN EACH ISLAND?</h2>
<HR>

In [31]:
penguin[penguin['sex'] == 'female']['island'].value_counts()

island
Biscoe       80
Dream        61
Torgersen    24
Name: count, dtype: int64

<h3>INSIGHT: THEREFORE, 80 FEMALE PENGUINS LIVE IN BISCOE, 61 IN DREAM AND 24 IN TORGERSEN</h3>
<hr>

<h3>QUESTION#9: HOW MANY MALE PNGUINS LIVE IN EACH ISLAND?</h3>
<hr>

In [32]:
penguin[penguin['sex'] == 'male']['island'].value_counts()

island
Biscoe       83
Dream        62
Torgersen    23
Name: count, dtype: int64

<h3>INSIGHT: THEREFORE, 83 MALE PENGUINS LIVE IN BISCOE, 62 IN DREAM AND 23 IN TORGERSEN</h3>
<hr>

<h3>QUESTION#10: WHAT GENDER OF PENGUIN COMMONLY FOUND IN EACH ISLAND?</h3>
<hr>

In [34]:
penguin.groupby(['island', 'sex']).size()

island     sex   
Biscoe     female    80
           male      83
Dream      female    61
           male      62
Torgersen  female    24
           male      23
dtype: int64

<h3>INSIGHT: THEREFORE, IN BISCOE MALE IS MORE COMMONLY FOUND AS WELL AS DREAM WHILE IN TORGERSEN FEMALE IS COMMONLY FOUND THE DIFFERENCE IS JUST 1-3 PENGUIN</h3>
<hr>

<h3>QUESTION#11: Which sex (male or female) tends to have the greater body mass on average?</h3>
<hr>

In [39]:
penguin[penguin['sex'] == 'female']['body_mass_g'].mean()

np.float64(3862.2727272727275)

In [40]:
penguin[penguin['sex'] == 'male']['body_mass_g'].mean()

np.float64(4545.684523809524)

<h3>INSIGHT: MALE TEND TO HAVE GREATER BODY MASS BASED ON THEIR AVERAGE </h3>
<hr>

<h3>QUESTION#12: How many penguins in the dataset have a body mass greater than 5,000 grams?</h3>
<hr>

In [48]:
count = (penguin['body_mass_g'] > 5000).sum()

In [49]:
count

np.int64(61)

<h3>INSIGHT: THERE ARE 61 PENGUINS THAT HAS A BODY MASS GREATER THAN 5000 GRAMS </h3>
<hr>